Release 3.2

nachocodoner opened the pull request:
## Still in development

Current released version: 3.2.

To use it, please do a

``` sh
meteor update --release 3.2
```

``` sh
meteor create --release 3.2
```

This beta version includes:

- https://github.com/meteor/meteor/pull/13636
- https://github.com/meteor/meteor/pull/13615 and https://github.com/meteor/meteor/pull/13628
- https://github.com/meteor/meteor/pull/13603
- https://github.com/meteor/meteor/pull/13554
- https://github.com/meteor/meteor/pull/13604

Please, help on test these since they are candidates for 3.2. Provide any feedback to the authors on those PRs.

All the rest PRs included: https://github.com/meteor/meteor/pulls?q=is%3Apr+milestone%3A%22Release+3.2%22

Seb-Dion committed:
Fixed issue with Firefox IOS not being marked as modern browser (#13126)

cunneen committed:
feat(logic-solver): :zap: recompile minisat with ALLOW_MEMORY_GROWTH option

Fixes issue #12902 with minisat: "Cannot enlarge memory arrays". Recompiled minisat.js as per https://github.com/cunneen/minisat/commit/9068151

Addresses #12902

vparpoil committed:
Update accounts-password to use argon2 instead of bcrypt

Users that have a bcrypt password stored get progressively migrated to argon2 when their password is checked by `checkPasswordAsync`

vparpoil committed:
Accounts-passowrd: Adding a test case to ensure smooth migration from bcrypt to argon2

vparpoil committed:
account-password: bump version to 3.0.4

vparpoil committed:
accounts-password: major version and more documentation about password encryption

vparpoil committed:
enable using other argon2 settings

vparpoil committed:
remove old comments (replaced by jsdoc)

vparpoil committed:
update tests for argon2 settings

nachocodoner committed:
Merge branch 'devel' into feature/argon2-password-encryption

vparpoil committed:
feat: add feature toggle argon2Enabled

vparpoil committed:
fix: remove console log of Urls when Meteor.isPackageTest

vparpoil committed:
fix: rename hashPassword in test file to hashPasswordWithSha to avoid confusion with server side hashPassword used to encrypt password

vparpoil committed:
fix: revert to initial password_tests file

vparpoil committed:
feat: add tests for argon : migration, set password and argon parameters change

vparpoil committed:
feat: enable rollback from argon2 to bcrypt

vparpoil committed:
Merge remote-tracking branch 'origin/devel' into feature/argon2-password-encryption

vparpoil committed:
adding the feature toggle to the docs

StorytellerCZ committed:
Update Facebook API version to v22

cunneen committed:
perf(logic-solver): :zap: minisat.js 236KB => 116KB
new minisat.js file, built from new meteor/minisat emscripten compiler optimizations, to reduce minisat.js file size from 236KB to 116KB .
Recompiled minisat.js as at https://github.com/cunneen/minisat/commit/8dc794e

Related to #12902 :"Cannot enlarge memory arrays"

gbuerk committed:
13599 - Removed the `waiting` variable in favor of using promises to keep track of fetch requests.  This was done to mitigate a race condition where a single fetch could increment `waiting` while the callback could be called twice, potentially decrementing  `waiting` to -1 leaving the driver stuck in `PHASE.FETCHING`

nachocodoner committed:
Merge remote-tracking branch 'vparpoil/feature/argon2-password-encryption' into argon2-support

nachocodoner committed:
Merge branch 'devel' into feature/argon2-password-encryption

nachocodoner committed:
fix waitUntil scenario

nachocodoner committed:
add waitUntil on brypt and argon2 migration to solve flakiness on slower execution contexts

nachocodoner committed:
Merge branch 'devel' into argon2-support

gbuerk committed:
Updates after code review to try to align closer with the original intent

gbuerk committed:
Merge branch 'devel' into fix-oplog-stuck-in-fetching-race-condition

nachocodoner committed:
specify alpha version for new feature

nachocodoner committed:
Merge remote-tracking branch 'origin/argon2-support' into argon2-support

nachocodoner committed:
fix versions for next alpha

nachocodoner committed:
update minisat to custom generated code

nachocodoner committed:
update minisat to custom generated code

nachocodoner committed:
fix code styling on affected files

nachocodoner committed:
Merge branch 'devel' into minisat-allow-memory-growth

StorytellerCZ committed:
Merge branch 'devel' into feature/arm64-minisat-allow-memory-growth

nachocodoner committed:
Merge branch 'devel' into argon2-support

nachocodoner committed:
Merge branch 'devel' into feature/arm64-minisat-allow-memory-growth

nachocodoner committed:
Merge branch 'devel' into minisat-allow-memory-growth

nachocodoner committed:
Merge pull request #13607 from meteor/argon2-support

Argon2 support

nachocodoner committed:
Merge pull request #13608 from meteor/minisat-allow-memory-growth

Recompile minisat with ALLOW_MEMORY_GROWTH option

nachocodoner committed:
bump for a new dev_bundle

nachocodoner committed:
re-run checks

nachocodoner committed:
Meteor version to 3.2.0-beta.0 :comet:

nachocodoner committed:
Meteor version to 3.2.0-beta.1 :comet:

nachocodoner committed:
Meteor version to 3.2.0-beta.1 :comet:

julio-rocketchat committed:
Update package.js to bump postcss

nachocodoner committed:
ensure ignore mongo to avoid annoying non-critical waring

nachocodoner committed:
ensure ignore mongo to avoid annoying non-critical waring

nachocodoner committed:
ensure some stage profiling are considered async to keep track of them

gbuerk committed:
Merge branch 'devel' into fix-oplog-stuck-in-fetching-race-condition

StorytellerCZ committed:
Node 22.14.0

StorytellerCZ committed:
Update included MongoDB to v7.0.16

Grubba27 committed:
FEATURE: add `shouldShowTime`

Relates to [this forum request](https://forums.meteor.com/t/suggestions-for-logging-in-galaxy/63119/6?u=grubba).

Those that use `colored-text` output format can now decide if they want the timestamp or can rely on their server/cloud environments info.

Grubba27 committed:
TEST: Add testing for `shouldShowTime` variable and run formatter in file

Grubba27 committed:
DEV: Add `shouldShowTime` type to Log namespace

Grubba27 committed:
DOCS: Add options to logging package

Grubba27 committed:
DOCS: update keys formatting in logging.md

Grubba27 committed:
DEV: rename shouldShowTime to `showTime`

nachocodoner committed:
patch bundle_version

nachocodoner committed:
Revert "patch bundle_version"

This reverts commit 89aa32dcb2bb87df1c6f0cc32dc2240fe28f4b1b.

nachocodoner committed:
patch bundle_version

nachocodoner committed:
patch bundle_version

nachocodoner committed:
re-run checks

nachocodoner committed:
re-run checks

StorytellerCZ committed:
Merge branch 'devel' into feature/facebook-api-v22

dr-dimitru committed:
fix: typo in link to Mongo-Cursor-forEachAsync

nachocodoner committed:
Merge pull request #13616 from meteor/dev-bundle-22.14.0.0

Node 22.14.0 & Mongo v7.0.16

nachocodoner committed:
Merge pull request #13622 from dr-dimitru/patch-1

fix: typo in link to Mongo-Cursor-forEachAsync

nachocodoner committed:
Merge branch 'release-3.2' into fix-profiling-stages

StorytellerCZ committed:
Update elliptic in node-stubs

julio-rocketchat committed:
Bump body-parser from 1.19.0 to 1.20.3

nachocodoner committed:
support async context on profiler async actions

nachocodoner committed:
correct async support on profiler

nachocodoner committed:
reuse meteor tool als context

nachocodoner committed:
avoid issue that runs the profiler function twice

nachocodoner committed:
avoid issue to properly finish on sync and async cases

nachocodoner committed:
fix require error

gbuerk committed:
Handling allSettled result better

nachocodoner committed:
dont mutate current entry

nachocodoner committed:
reuse existing async local storage context

nachocodoner committed:
use const

nachocodoner committed:
Merge branch 'release-3.2' into fix-oplog-stuck-in-fetching-race-condition

Grubba27 committed:
Merge pull request #13618 from meteor/feature/add-option-to-remove-timestamp-in-logging

FEATURE: add `shouldShowTime` to `logging` package

zodern commented on packages/accounts-password/package.js:
```diff
@@ -5,11 +5,12 @@ Package.describe({
   // during the Meteor 1.5.1 release process, so versions 2.0.0-beta.2
   // through -beta.5 and -rc.0 have already been published.
-  version: "3.0.3",
+  version: "4.0.0-beta320.1",
```
Why does adding argon2 require a major version increase? The change log doesn't mention any breaking changes.

StorytellerCZ replied:
The original implementation was breaking, but now we can get away with just a feature version since Argon2 is hidden behind a flag.

nachocodoner replied:
Yes, the reason was at the beginning it was a version with major breaking changes. Not anymore, so I will respect a minor version bump.

nachocodoner committed:
remove als initialization as part of server startup

nachocodoner committed:
Merge pull request #13628 from meteor/support-async-profiler

Support async profiler actions

nachocodoner committed:
Merge pull request #13615 from meteor/fix-profiling-stages

Fix Meteor 3 profiler async actions and bundle stages not profiling as expected

nachocodoner committed:
Merge pull request #13632 from meteor/gbuerk-fix-oplog-stuck-in-fetching-race-condition

[Contribution] Fix oplog stuck in fetching race condition

nachocodoner committed:
Merge pull request #13614 from meteor/skel-meteorvite-warn

Ignore meteor/mongo to avoid annoying non-critical waring

nachocodoner committed:
implement meteor benchmark cmd

nachocodoner committed:
improve command with bash live feedback and ensure expected git version

nachocodoner committed:
dont print bashLive input

nachocodoner committed:
improve bashLive outputting by perserve formatting

nachocodoner committed:
Merge pull request #13625 from meteor/feature/npm-stubs-security-update

Update elliptic in node-stubs

Grubba27 committed:
DEV: bump meteor-node-stubs to 1.2.13

nachocodoner committed:
adjust name of the command

nachocodoner committed:
add help content for new profile command

nachocodoner committed:
better formatting for help

nachocodoner committed:
better context

StorytellerCZ committed:
Changelog update for meteor-node-stubs

nachocodoner committed:
Merge pull request #13626 from julio-rocketchat/bump-body-parser-in-oauth

Bump body-parser from `1.19.0` to `1.20.3` in the `oauth` package

nachocodoner committed:
Merge pull request #13613 from julio-rocketchat/bump-postcss

Bump postcss in minifier-css to fix vulnerabilities

nachocodoner committed:
Merge remote-tracking branch 'sebdion/fix-13126-setMinimumBrowserVersions' into Seb-Dionfix-13126-setMinimumBrowserVersions

# Conflicts:
#	packages/modern-browsers/modern.js

nachocodoner committed:
fix format

nachocodoner committed:
reformat with eslint

nachocodoner committed:
Merge pull request #13640 from meteor/Seb-Dionfix-13126-setMinimumBrowserVersions

fix 13126 set minimum browser versions (Contribution)

nachocodoner committed:
Merge pull request #13636 from meteor/meteor-benchmark-cmd

Meteor bundler profile command

nachocodoner committed:
Merge pull request #13600 from meteor/feature/facebook-api-v22

Update default Facebook API version to v22

nachocodoner committed:
Merge branch 'devel' into release-3.2

nachocodoner committed:
bump version to generate a new dev_bundle

nachocodoner committed:
Meteor version to 3.2-beta.2 :comet:

StorytellerCZ commented on packages/oauth/package.js:
```diff
@@ -37,7 +37,7 @@ Package.onUse(api => {
 
 Npm.depends({
-  'body-parser': '1.19.0',
+  'body-parser': '1.20.3',
```
If #13323 as whole is not acceptable, then I think we can cherry-pick the part that just removes this dependency as we have this in Express.

nachocodoner replied:
It's not that it's unacceptable, but it involves different changes and context than the contributions we merged for 3.2. We need time to analyze and test the affected parts, including any side effects. Especially since some areas on those changes may not have full test coverage. Deferring some changes helps us focus our efforts.

That PR could be a good candidate for the first beta of 3.3 (or next 3.2.1 patch, undecided yet). As with 3.2, this would give time to everyone for testing, including our final verifications before the official release. But we also need to find time to understand each change and the purpose first.

nachocodoner replied:
Btw, I have a list of the PRs you attach in release milestones, including that one, so we can review and decide what to include in future releases. We plan to do this for each release based on the challenges on those, the testing capacity on us and the community to test before official. Keep flagging issues in milestones for next releases, some areas we may miss to track, and those warnings help a lot.

StorytellerCZ replied:
Yes, feel free to move the issues/PRs to their appropriate release milestones. We should probably jump on a call/stream sometime and do a few together and talked about this so that I can better triage.

vparpoil committed:
fix: argon2 default parameters were too strong

vparpoil committed:
Merge remote-tracking branch 'origin/devel' into feature/argon2-password-encryption

nachocodoner committed:
Merge branch 'release-3.2' into feature/argon2-password-encryption

nachocodoner committed:
Merge branch 'release-3.2' into feature/arm64-minisat-allow-memory-growth

nachocodoner committed:
support meteor client and server entrypoint custom envs

nachocodoner committed:
support METEOR_LOG_DIR env config

nachocodoner committed:
Merge pull request #13554 from vparpoil/feature/argon2-password-encryption

Feature argon2 password encryption

nachocodoner committed:
Merge pull request #13604 from cunneen/feature/arm64-minisat-allow-memory-growth

feat(logic-solver): ⚡️ recompile minisat with ALLOW_MEMORY_GROWTH option

nachocodoner committed:
Merge branch 'devel' into release-3.2

nachocodoner committed:
ensure options to handle size

nachocodoner committed:
update cmd to add size options

nachocodoner committed:
update cmd to add size options

nachocodoner committed:
update cmd to add size options

nachocodoner committed:
add option to skip cache

nachocodoner committed:
update docs for meteor profile cmd

nachocodoner committed:
better --size-only option

nachocodoner committed:
tweak to version branch

nachocodoner committed:
bump for a new dev_bundle

nachocodoner committed:
Merge pull request #13656 from meteor/profile-cmd-feedback

Address `meteor profile` feedback

nachocodoner committed:
re-run checks

nachocodoner committed:
Meteor version to 3.2 :comet:

nachocodoner committed:
update installer to meteor 3.2 release

italojs committed:
feat: native profile inspector
- Enhance profiling capabilities with inspector support and update documentation.
- Added .cursorignore and .cursorrules to .gitignore.
- Updated PERFORMANCE.md and README.md with detailed instructions for using inspector profiling, including environment variable configurations and output handling.
- Improved profiling function to manage inspector sessions and handle large profile sizes effectively.

italojs committed:
log: adding –max_old_space_size warn

italojs committed:
refactor: streamline profiling termination handling and cleanup

- Consolidated termination logic for profiling into a single function to reduce redundancy.
- Removed unnecessary comments and code related to heap limit warnings.
- Improved clarity in handling synchronous results during profiling completion.
- Enhanced error logging for profiling termination events.

Grubba27 committed:
Merge branch 'devel' into release-3.2

nachocodoner committed:
remove meteor profile docs in favor of improved version
